### PR TITLE
Validate sprayed rooms against found rooms

### PR DESCRIPTION
**Pull request by vincent-karuri**
PR for issue #69.

### CODE REVIEW

File: opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java [hunk 1]
```diff
@@ -47,6 +53,8 @@ protected void attachLayout(String stepName, Context context, JsonFormFragment f
         }
         editText.setGravity(Gravity.START);
         editText.setSingleLine(false);
+
+        addRelativeNumericIntegerValidator(jsonObject, formFragment, editText);
```
Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java]: Move to native form library

File: opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java [hunk 2]
```diff
@@ -57,4 +65,24 @@ protected int getLayout() {
             return super.getLayout();
         }
     }
+
+    private void addRelativeNumericIntegerValidator(JSONObject editTextJSONObject, JsonFormFragment formFragment, MaterialEditText editText) throws JSONException {
```
Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java]: Move to native form libary

File: opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java (hunk 1 above)
Comment (vincent-karuri) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java]: Done.

Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java]: Move this method call to native form EditTextFactory 
com.vijay.jsonwizard.widgets.EditTextFactory#attachLayout

File: opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java (hunk 2 above)
Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java]: Move this method call to  nativeform EditTextFactory class

File: opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java (hunk 1 above)
Comment (vincent-karuri) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java]: Done.

File: opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java (hunk 2 above)
Comment (vincent-karuri) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java]: Done.

**Comment by githengi** [comment]
I think the value 99 is not being handled, maybe in the RelativeMaxNumericValidator one can specify a value that can be ignored during validation

**Comment by githengi** [comment]
Add relevance so that no. of sprayed rooms is shown after one inputs the number of rooms

**Comment by vincent-karuri** [comment]
> Add relevance so that no. of sprayed rooms is shown after one inputs the number of rooms

Done.

**Pull request merged**